src/impl/dtlssrtptransport.cpp

```diff
@@ -218,33 +218,51 @@ bool DtlsSrtpTransport::demuxMessage(message_ptr message) {
 	}
 }
 
+EncryptionParams DtlsSrtpTransport::getEnctyptionParams(const char *suite) {
+	if (strcmp(suite, "SRTP_AES128_CM_SHA1_80")==0)
+		return {SRTP_AES_128_KEY_LEN, SRTP_SALT_LEN, srtp_profile_aes128_cm_sha1_80};
+	if (strcmp(suite,"SRTP_AES128_CM_SHA1_32")==0)
+		return {SRTP_AES_128_KEY_LEN, SRTP_SALT_LEN, srtp_profile_aes128_cm_sha1_32};
+	if (strcmp(suite,"SRTP_AEAD_AES_128_GCM")==0)
+		return {SRTP_AES_128_KEY_LEN, SRTP_AEAD_SALT_LEN, srtp_profile_aead_aes_128_gcm};
+	if (strcmp(suite,"SRTP_AEAD_AES_256_GCM")==0)
+		return {SRTP_AES_256_KEY_LEN, SRTP_AEAD_SALT_LEN, srtp_profile_aead_aes_256_gcm};
+	throw std::logic_error("Unexpected SRTP suite name: " + std::string(suite));
+}
+
 void DtlsSrtpTransport::postHandshake() {
 	if (mInitDone)
 		return;
 
-	static_assert(SRTP_AES_ICM_128_KEY_LEN_WSALT == SRTP_AES_128_KEY_LEN + SRTP_SALT_LEN);
-
-	const size_t materialLen = SRTP_AES_ICM_128_KEY_LEN_WSALT * 2;
-	unsigned char material[materialLen];
+	auto profile = SSL_get_selected_srtp_profile(mSsl);
+	if (!profile)
+		throw std::runtime_error("Failed to get SRTP profile");
+	PLOG_DEBUG << "srtp profile used is: " << profile->name;
```

Comment:
This is for OpenSSL only and therefore must be in a `#if USE_GNUTLS [...] #else [...] #endif` block. For GnuTLS the correct approach is to call `gnutls_srtp_get_selected_profile()` but you may hardcode the `SRTP_AES128_CM_SHA1_80` profile instead.